servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -324,30 +329,33 @@ private static Status getStatusDetails(final HttpHeaders headers) {
     }
 
     /**
-     * Construct the gRPC header {@code grpc-accept-encoding} representation of the given codings.
+     * Construct the gRPC header {@code grpc-accept-encoding} representation of the given context.
      *
      * @param codings the list of codings to be used in the string representation.
-     * @return a comma separated string representation of the codings for use as a header value
+     * @return a comma separated string representation of the codings for use as a header value or {@code null} if
+     * the only supported coding is {@link Identity#identity()}.
      */
+    @Nullable
     private static CharSequence acceptedEncodingsHeaderValueOrCached(final List<ContentCodec> codings) {
         return ENCODINGS_HEADER_CACHE.computeIfAbsent(codings, (__) -> acceptedEncodingsHeaderValue0(codings));
     }
 
+    @Nullable
     private static CharSequence acceptedEncodingsHeaderValue0(final List<ContentCodec> codings) {
-        StringBuilder builder = new StringBuilder();
+        StringBuilder builder = new StringBuilder(codings.size() * (12 + CONTENT_ENCODING_SEPARATOR.length()));
         for (ContentCodec codec : codings) {
             if (codec == identity()) {
```

Comment:
Users may have their own version of "identity" codec. We should implement `equals` and `hashCode` for `IdentityContentCodec` then use `identity().equals(codec)` here and in `HeaderUtils` for HTTP.

Reply:
Agreed, issue: https://github.com/apple/servicetalk/issues/1455